SimpleInventoryManagementSystem/NoSqlDB.cs

```diff
@@ -0,0 +1,113 @@
+﻿using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace SimpleInventoryManagementSystem
+{
+    public class NoSqlDB : IDatabase
+    {
+
+        IMongoDatabase database = ConnectionInitializer.InitializeMongoConnection();
+
+
+        public void AddProduct(Product product)
+        {
+            try
+            {
+                var collection = database.GetCollection<Product>("products");
+                product.Id = MaxProductId() + 1;
+                collection.InsertOne(product);
+                Console.WriteLine("Product added successfully");
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error adding product: {ex.Message}");
+            }
+        }
+
+        public void DeleteProduct(Product product)
+        {
+            try
+            {
+                var collection = database.GetCollection<Product>("products");
```

Comment:
this is shared between methods, you can extract it and initialize in the constructor